nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,799 @@
+
+        y : array or list of length n_samples
+            The dependent variable (age, sex, QI, etc.).
+            Target variable to predict. Must have exactly as many elements as
+            3D images in niimg.
+
+        Attributes
+        ----------
+        `masker_` : instance of NiftiMasker
+            The nifti masker used to mask the data.
+
+        `mask_img_`: NiImage
+            Mask computed by the masker object.
+
+        `classes_`: numpy.ndarray
+            Classes to predict. For classification only.
+
+        `screening_percentile_` : float
+            Screening percentile corrected according to volume of mask,
+            relative to the volume of standard brain.
+
+        `coef_`: numpy.ndarray, shape=(n_classes, n_features)
+            Contains the mean of the models weight vector across
+            fold for each class.
+
+        `intercept_` : narray, shape (nclasses -1,)
+            Intercept (a.k.a. bias) added to the decision function.
+            It is available only when parameter intercept is set to True.
+
+        `cv_` : list of pairs of lists
+            List of the (n_folds,) folds. For the corresponding fold,
+            each pair is composed of two lists of indices,
+            one for the train samples and one for the test samples.
+
+        `std_coef_`: numpy.ndarray, shape=(n_classes, n_features)
+            Contains the standard deviation of the models weight vector across
+            fold for each class.
+
+        `coef_img_`: dict of NiImage
+            Dictionary containing `coef_` with class names as keys,
+            and `coef_` transformed in NiImages as values. In the case
+            of a regression, it contains a single NiImage at the key 'beta'.
+
+        `std_coef_img_`: dict of NiImage
+            Dictionary containing `std_coef_` with class names as keys,
+            and `coef_` transformed in NiImages as values. In the case
+            of a regression, it contains a single NiImage at the key 'beta'.
+
+        `cv_y_true_` : numpy.ndarray, shape=(n_samples * n_folds, n_classes)
+            Ground truth labels for left out samples in inner cross-validation.
+
+        `cv_y_pred_` : numpy.ndarray, shape=(n_samples * n_folds, n_classes)
+            Predicted labels for left out samples in inner cross-validation.
+
+        `cv_params_`: dict of lists
+            Best point in the parameter grid for each tested fold
+            in the inner cross validation loop.
+
+        `cv_indices_` : numpy.ndarray, shape=(n_samples * n_folds)
+            Indices of the inner cross-validation folds.
+
+        `cv_scores_` : dict, (classes, n_folds)
+            Scores (misclassification) for each parameter, and on each fold
+        """
+        # Setup memory, parallel and masker
+        if isinstance(self.memory, _basestring) or (self.memory is None):
+            self.memory_ = Memory(cachedir=self.memory, verbose=self.verbose)
```

Comment:
Why do we take memory as an attribute ? It is not documented as not used anywhere ?